scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
In Python 2 when unicode URL is passed it is converted to bytes using utf-8 encoding, regardless of `encoding` passed. Should we use `encoding` instead? If so, the warning may become misleading. There should be a warning if encoding is not passed, but if user passes it explicitly then it may be fine to pass unicode URL in Python 2.